### ISSUE TITLE
Pronominal adverb and clausal argument (?) in German

### ISSUE THREAD

**Issue by aluecking** [UniversalDependencies/docs#840]
I'm looking at UD annotations for German sentences and I frequently come across structures such as "dadurch ... dass", "damit ... dass", "dafür ... dass". Now I wonder how to capture them in UD. The first example from grew from UD_German-HDT@2.9 is "er rechne damit, dass ...". 
http://match.grew.fr/?corpus=UD_German-HDT@2.9&custom=61dc77e69d543
In this example, the main verb "rechne" receives two arguments: the OBJ "damit" and the "dass ..." sentence as CCOMP. I would assume, however, that the frame of "rechnen" is already satisfied by just one of these arguments. For instance, you can not have a PP argument and an additional CCOMP: *"Er rechnet mit gutem Wetter, dass es nicht regnet".

"dadurch ... dass" is treated differently (same resource): 
http://match.grew.fr/?corpus=UD_German-HDT@2.9&custom=61dc7560210b5
Here, "dadurch" is merely an ADVMOD which remains unconnected to the "dass" sentence. 

My inclination in both cases is, to conceive the pro form as OBJ of the main verb, and the "dass" sentence as the clausal argument of the pro form, probably ACL. 

Are there any ideas or previous discussion on this? 

Many thanks! Best,
Andy

**Comment by gossebouma** [comment]
I would agree with you that you want to analyze these as a unit, and not as two independent dependents of the verb. However, it might be problematic to label the adverb "damit" as OBJ as in cases like "Er rechnet mit gutem Wetter", "wetter" would be an OBL dependent. It seems logical to have the adverb as OBL as well, but I think the validation rules might not allow for this. So, in examples like "Er rechnet damit", "damit" is an ADVMOD. One might argue that in spite of the fact that these words are UPOS=ADV, they really are similar to PPs, and thus should also be labeled as OBL. 

That being said, in the Dutch data yet another solution is chosen. In comparable cases with prepositional adverb "erop", the complement clause is a CCOMP dependent of the verb and the adverb has the 'case' relation.  This query gives some examples with "erop":	http://match.grew.fr/?corpus=UD_Dutch-Alpino@2.9&custom=61e692c0cbe12

To me this also seems like a suboptimal solution, probably motivated by validation rules. My preferred analysis would be to have the prepositional adverb as OBL dependent of the verb, and the clausal complement as CCOMP dependent of the prepositional adverb.

**Comment by martinpopel** [comment]
> It seems logical to have the adverb as OBL as well, but I think the validation rules might not allow for this

Yet another option would be to treat "damit" (and dadurch, dafür,...) as a multi-word token da+mit, where "mit" would be a standard adposition and then the choice deprel=obl would be natural.
I am **not proposing** this option as I don't have enough knowledge (about German, user expectations etc). I am just mentioning it in this discussion for completeness.

**Comment by amir-zeldes** [comment]
If we disregard conflicts with STTS-style tokenization, then yes, I agree the expected UD behavior is probably to split da+mit as a multiword token. In that case, since the clause saturates the exact same argument as "damit", I would have treated the pronominal part as an expletive:

  * Er rechnet damit, dass sie kommen

expl(rechnet,da)
case(da, mit)
ccomp(rechnet, kommen)

But if MWTs are not feasible here due to legacy tokenization, then the entire damit can also be seen as an expletive, leaving the case relation unexpressed.

**Comment by gossebouma** [comment]
In Dutch, the expletive 'er' and preposition can actually be split, as in: 

_Tinbergen is er heilig van overtuigd dat oost en west naar elkaar toegroeien (Tinbergen is deeply convinced that east and west move toward eachother)_

but both the expl and the prep are dependents of the ccomp head:

expl(toe_groeien,er)
case(toe_groeien,van)
ccomp(overtuigen,toe_groeien) 

(a few more [examples](http://match.grew.fr/?corpus=UD_Dutch-Alpino@2.9&custom=61e7bfb72917f) )

This seems like a desperate attempt to keep the three elements together. The usual analysis of constructions is that the ccomp cannot fill the position filled by an NP in normal PP constructions, and so it is 'extraposed', with the expletive filling the argument role of the preposition (which for some reason then also cannot follow the preposition and so is 'fronted') (very loosely trying to recall the classical study by Jan van Riemsdijk, a case study in syntactic markedness). Translating this to UD terms, that would give some credibility to making the preposition a dependent of the ccomp head. I am less sure about the status of the expletive. 

Re splitting the prepositional adverbs: that would be nice as it would make the same analysis possible, but for the Dutch treebanks we have never had such cases and I am not sure what it might break.

**Comment by dan-zeman** [comment]
@gossebouma is right that in _Er rechnet mit gutem Wetter_ "He counts on good weather", _Wetter_ should be `obl`. Therefore, if a clause commutes with a _mit_-marked nominal, it should be `advcl`, not `ccomp`.

However, here I wouldn't attach the clause directly to the verb _rechnet_. Instead I would say that it modifies _damit_. The same approach is currently taken in a number of UD languages, e.g. Czech (http://hdl.handle.net/11346/PMLTQ-PQFM), Slovak (http://hdl.handle.net/11346/PMLTQ-NOIX), Upper Sorbian (http://hdl.handle.net/11346/PMLTQ-ESJA), Polish (http://hdl.handle.net/11346/PMLTQ-OP9Z), Russian (http://hdl.handle.net/11346/PMLTQ-7XGG), Belarusian (http://hdl.handle.net/11346/PMLTQ-VPTZ), Ukrainian (http://hdl.handle.net/11346/PMLTQ-BMJ9), Slovenian (http://hdl.handle.net/11346/PMLTQ-EDHW), Croatian (http://hdl.handle.net/11346/PMLTQ-KUXD), Serbian (http://hdl.handle.net/11346/PMLTQ-4Q2O), or Bulgarian (http://hdl.handle.net/11346/PMLTQ-ZEFM).

One difference is that in the Slavic languages the case-bearing element is a demonstrative pronoun or determiner, hence the subordinate clause is `acl`. In German the pro-forms are tagged as demonstrative adverbs, so the clause should be `advcl`.

**Comment by gossebouma** [comment]
I agree that advcl is a better choice. (The ccomp decision was probably due to the PP complement having more or less complement status (ie cannot easily be omitted without changing the meaning of the verb) and is often seen as such in traditional grammar). Since we do not give regular PPs complement status in UD, we should not either for these cases.

Attaching it to the adverb 'damit' also sounds fine. But then I am still not exactly sure how to deal with cases (in Dutch) where the 'er' and the preposition are split. Like this? ("er" is UPOS PRON) 

Ik reken er niet op dat hij komt
I  count EXPL not on that he comes

advcl(er,komt)
case(er,op)
obl(reken,er)

**Comment by dan-zeman** [comment]
Would it be too radical to tag _er_ `PRON` instead of `ADV`?

**Comment by amir-zeldes** [comment]
What I find a little odd about this analysis is that it suggests "er" has a modifier clause itself, whereas actually it's the verbs valency that is responsible for the advcl happening. Shouldn't "er" still be `expl`? Isn't that what we do when the subject or object is recapitulated by a clause, but we get a dummy pronoun in matrix position? Consider this example from Alpino:

  * zij maakten het/expl zo leuk mogelijk om te komen/ccomp
  * They made it/expl as fun as possible to come/ccomp

In other words "~they made to come as fun as possible". Isn't this the same but in oblique, and if so, shouldn't we expect `expl` here too?

**Comment by dan-zeman** [comment]
> What I find a little odd about this analysis is that it suggests "er" has a modifier clause itself, whereas actually it's the verbs valency that is responsible for the advcl happening.

That's why the Slavic analysis makes perfect sense in the Slavic languages (if we ignore the fact that UD mostly avoids "spoiling" its guidelines by valency), because only the demonstrative has the required case marking that allows it to fill the slot; and since the slot is already filled, it is modified in a rather uniform way, which would stay the same if the demonstrative was in a different valency slot (including the subject) instead. I cannot judge to what extent the same analysis can be applied to the Dutch _er_; but given my background bias, this would be the first solution I'd think of.

**Comment by gossebouma** [comment]
> Would it be too radical to tag _er_ `PRON` instead of `ADV`?

'er' is a PRON already ;-)

**Comment by gossebouma** [comment]
> What I find a little odd about this analysis is that it suggests "er" has a modifier clause itself, whereas actually it's the verbs valency that is responsible for the advcl happening. 

That is the reasoning behind the current analysis, where the advcl was attached to the verb, and 'er' and the prep are EXPL and CASE dependents of the head of the advcl. But I think the proposal of Dan captures the fact that er+P+clause form a unit, even if discontinuous (and this is how it is analysed in traditional grammar)

**Comment by dan-zeman** [comment]
> > Would it be too radical to tag _er_ `PRON` instead of `ADV`?
> 
> 'er' is a PRON already ;-)

Is it? When searching Alpino 2.9 (http://hdl.handle.net/11346/PMLTQ-XE61), I found 185 adverbs.

**Comment by gossebouma** [comment]
In cases where 'er' has deprel expl, the upos=PRON, in all other cases (mostly advmod, obl) it is ADV.  

In the underlying annotation it is always a pronoun (VNW) but that will not work for the advmod  cases I assume. The obl cases should probably be PRON as well, if we want to keep some parallelism with the construction where there is also an adverbial clause. There are also rare cases where 'er' is nsubj or obj, in those cases I think it should also be PRON. 

So I agree the picture is a bit messy here and needs work.

**Comment by rueter** [comment]
Hi, @aluecking!
I was glad to see that @amir-zeldes provided a simple and illustrated sentence from German.
Instead of treating the construction as one containing an expletive, I will look at the same construction in Estonian.
> 
>     * Er rechnet damit, dass sie kommen
> 
*Tema    arvestab      sellega , et      nad   tulevad.
Sg3        count.Sg3    it.Com ,  that  Pl3    come.Pl3
(The ordering of the German and Estonian are matched)

nsubj(arvestab, Tema)
obl(arvestab, sellega)
acl(sellega, tulevad)
nsubj(tulevad, nad)
mark(tulevad, et)

In Estonian, we see a comitative case <-ga>, which has the same meaning as the German <mit>. As <sellega> is considered a regular demonstrative pronoun, the dependency is obl.

The second phrase <et nad tulevad> modifies the demonstrative pronoun <sellega> with an acl.

Since German treats <damit> as an advmod, I would analogically use an advcl and draw the following dependencies.

* Er rechnet damit, dass sie kommen
nsubj(rechnet, Er)
advmod(rechnet, damit)
advcl(damit, kommen)
nsubj(kommen, sie)
mark(kommen, dass)

cf. Estonian-EWT
ewtb2_149414_36
Finnish has the same kinds of structures.

North_Sami-Giella also has this type of structure:
id="1159" 
Kvaliteahtta sihkkarasto      dainna      , ahte   bargit      dovdet       kultuvrra.
quality          is.guarantied   therewith ,  that   workers   know.Pl3   culture.Acc
In North Sami, however, what would appear to be a regular demonstrative pronoun in the comitative <dainna> is coded as ADV.
and the relations
nsubj(sihkkarasto, Kvaliteahtta)
advmod(sihkkarasto, dainna)
advcl(sihkkarasto, dovdet)
nsubj(dovdet, bargit)
obj(dovdet, kultuvrra)
mark(dovdet, ahte)

I am perplexed by, as to why the obl and acl dependencies are NOT
parallel to the advmod and advcl.

**Comment by dan-zeman** [comment]
> In cases where 'er' has deprel expl, the upos=PRON, in all other cases (mostly advmod, obl) it is ADV.
> 
> In the underlying annotation it is always a pronoun (VNW) but that will not work for the advmod cases I assume. The obl cases should probably be PRON as well, if we want to keep some parallelism with the construction where there is also an adverbial clause. There are also rare cases where 'er' is nsubj or obj, in those cases I think it should also be PRON.
> 
> So I agree the picture is a bit messy here and needs work.

Yes, `obl` cases should be `PRON`. And since `advmod` is in UD essentially equivalent of `obl` for the case that the word is `ADV` rather than a nominal, I don't see a problem in making them all `PRON` and `obl`, if it is in line with the native analysis.

**Comment by aluecking** [comment]
The Estonian construction looks quite good. But I agree:

> Since German treats as an advmod, I would analogically use an advcl

So I think ADVMOD+ADVCL seems to be the best working option (for the time being, at least).
